### PR TITLE
Fixed NewServer() init logic

### PR DESCRIPTION
**Pull request by UladzimirTrehubenka**
### 1. What does this pull request do?
Fixed NewServer() init logic
### 2. Which issues (if any) are related?
https://github.com/coredns/coredns/issues/1636
### 3. Which documentation changes (if any) need to be made?
none

### CODE REVIEW

File: core/dnsserver/server.go [hunk 1]
```diff
@@ -64,16 +64,19 @@ func NewServer(addr string, group []*Config) (*Server, error) {
 		// set the config per zone
 		s.zones[site.Zone] = site
 		// compile custom plugin for everything
+		registered := false
 		if site.registry != nil {
```
Comment (miekg) [inline core/dnsserver/server.go]: can we just re use `site.registry`?

Comment (rdrozhdzh) [reply core/dnsserver/server.go]: @miekg, actually, there is no problem with reusing site.registry. The problem is with initializing fields Server.trace and Server.classChaos which should be done for each Server object, even if they are reusing the same config, e.g. see line https://github.com/coredns/coredns/pull/1637/files#diff-a277767ea17c7dfa10555939729f0d65R90

Comment (UladzimirTrehubenka) [reply core/dnsserver/server.go]: Reimplemented with using site.registry

**Comment by miekg** [comment]
thanks!

**Pull request merged**